src/dtos/revision-dto.ts

```diff
@@ -0,0 +1,55 @@
+import { Revision } from '../entities/revision';
+import { FileImport } from '../entities/file-import';
+
+import { ImportDTO } from './fileimport-dto';
+
+export class RevisionDTO {
+    id: string;
+    revision_index: number;
+    creation_date: string;
+    previous_revision_id?: string;
+    online_cube_filename?: string;
+    publish_date?: string;
+    approval_date?: string;
+    approved_by?: string;
+    created_by: string;
+    imports: ImportDTO[];
+    dataset_id?: string;
+
+    static async fromRevision(revision: Revision): Promise<RevisionDTO> {
+        const revDto = new RevisionDTO();
+        revDto.id = revision.id;
+        revDto.revision_index = revision.revisionIndex;
+        revDto.dataset_id = (await revision.dataset).id;
+        revDto.creation_date = revision.creationDate.toISOString();
+        revDto.previous_revision_id = (await revision.previousRevision)?.id || '';
+        revDto.online_cube_filename = revision.onlineCubeFilename;
+        revDto.publish_date = revision.publishDate?.toISOString() || '';
+        revDto.approval_date = revision.approvalDate?.toISOString() || '';
```

Comment:
Same as above, don't stick empty strings in if we mean undefined.

Reply:
Fixed this.